### PR TITLE
fix: add isDeepLinking state to prevent deeplink lock

### PR DESCRIPTION
**Pull request by beths-ledger**
### 📝 Description

EARN dashboard on Android deep links (e.g. on tapping "Get [coin]" button) were triggering the LLM password lock.`<SkipLock />` does not work in this case because the background state is triggered more than once.

- Adds `isDeepLinking` to state
- Set on when deep link initiated
- Reset to false when earn screen is returned to, or drawer is closed

### ❓ Context

- **Impacted projects**: `` ledger-live-mobile
- **Linked resource(s)**: `` https://ledgerhq.atlassian.net/browse/LIVE-9180

### ✅ Checklist

- [ ] **Test coverage** 
- [x] **Atomic delivery** 
- [x] **No breaking changes** 

### 🚀 Expectations to reach

_Please make sure you follow these [**Important Steps**](https://github.com/LedgerHQ/ledger-live/blob/develop/CONTRIBUTING.md#important-steps)._

_Pull Requests must pass the CI and be internally validated in order to be merged._

### CODE REVIEW

File: apps/ledger-live-mobile/src/reducers/types.ts [hunk 1]
```diff
@@ -268,6 +268,8 @@ export type EarnState = {
     message?: string;
     messageTitle?: string;
   };
+  /** Earn web player deeplinks need to skip privacy lock. Resets on return to Earn dashboard. If other deeplinks also need to skip lock, could move this to AppState or somewhere more global. */
```
Comment (mitchellv-ledger) [inline apps/ledger-live-mobile/src/reducers/types.ts:271]: I think best to move directly to appstate

**Review by mitchellv-ledger — CHANGES_REQUESTED** [review]
Looks great, I think the redux state should live in appstate instead of earn as it's not earn's responsibility to maintain this feature

File: apps/ledger-live-mobile/src/reducers/types.ts (hunk 1 above)
Comment (beths-ledger) [reply apps/ledger-live-mobile/src/reducers/types.ts:271]: i actually agree with this, thanks for the feedback!

File: apps/ledger-live-mobile/src/components/RootNavigator/EarnLiveAppNavigator.tsx [hunk 2]
```diff
@@ -23,11 +24,12 @@ type NavigationProps = BaseComposite<
 >;
 
 const Earn = (props: NavigationProps) => {
+  const dispatch = useDispatch();
   // Earn dashboard feature flag
   const ptxEarn = useFeature("ptxEarn");
   const paramAction = props.route.params?.action;
   const navigation = props.navigation;
-  const accounts = useSelector(accountsSelector);
+  const accounts = useSelector(shallowAccountsSelector);
```
Comment (ofreyssinet-ledger) [inline apps/ledger-live-mobile/src/components/RootNavigator/EarnLiveAppNavigator.tsx:32]: is this change intentional in the scope of this PR?

**Review by mitchellv-ledger — APPROVED** [review]
awesome 🏆
